components/QueryControls/index.tsx

```diff
@@ -0,0 +1,274 @@
+            <RangeControl
+              label={__('Number of Posts', 'wp-curate')}
+              help={__('The maximum number of posts to show.', 'wp-curate')}
+              value={numberOfPosts}
+              onChange={setNumberOfPosts}
+              min={minNumberOfPosts}
+              max={maxNumberOfPosts}
+            />
+          ) : null}
+          <RangeControl
+            label={__('Offset', 'wp-curate')}
+            help={__('The number of posts to pass over.', 'wp-curate')}
+            onChange={(newValue) => setAttributes({ offset: newValue })}
+            value={offset}
+            min={0}
+            max={20}
+          />
+        </PanelBody>
+
+        <PanelBody
+          title={__('Select Posts', 'wp-curate')}
+          initialOpen={false}
+          className="manual-posts"
+        >
+          {manualPosts.map((_post, index) => (
+            <PanelRow
+              // eslint-disable-next-line react/no-array-index-key
+              key={index}
+              className={classnames(
+                'manual-posts__container',
+                { 'manual-posts__container--selected': manualPosts[index] },
+              )}
+            >
+              <span className="manual-posts__counter">{index + 1}</span>
+              <PostPicker
+                allowedTypes={allowedPostTypes.map((type) => type.slug)}
+                onReset={() => setManualPost(0, index)}
+                onUpdate={(id: number) => { setManualPost(id, index); }}
+                value={manualPosts[index] || 0}
+                className="manual-posts__picker"
+              />
+            </PanelRow>
+          ))}
+        </PanelBody>
+
+        <PanelBody
+          title={__('Query Parameters', 'wp-curate')}
+          initialOpen={false}
+        >
+          <Checkboxes
+            label={__('Post Types', 'wp-curate')}
+            value={postTypes}
+            onChange={(newValue) => setAttributes({ postTypes: newValue, backfillPosts: [] })}
+            options={displayTypes}
+          />
+          {allowedTaxonomies.map((taxonomy) => (
+            <Fragment key={taxonomy.slug}>
+              { /* @ts-ignore */ }
+              <TermSelector
+                label={taxonomy.name}
+                subTypes={[taxonomy.slug]}
+                selected={terms[taxonomy.slug] ?? []}
+                onSelect={(newCategories: Term[]) => setTerms(taxonomy.slug, newCategories)}
+                multiple
+              />
+              {terms[taxonomy.slug]?.length > 1 ? (
+                <SelectControl
+                  label={sprintf(
+                    __('%s Relation', 'wp-curate'),
+                    taxonomy.name,
+                  )}
+                  help={__('AND: Posts must have all selected terms. OR: Posts may have one or more selected terms.', 'wp-curate')}
+                  options={andOrOptions}
+                  onChange={(newValue) => setTermRelation(taxonomy.slug, newValue)}
```

Comment:
We generally standardize on `next` as the new value, and there's a mix of both in this file.